File: basic data structure code in c/sorting/quick_sort.c
#include<stdio.h>

int partition(int[],int ,int);
void quicksort(int[],int,int);
void swap(int arr[],int a,int b);

void main(){
		int arr[6]={3,6,5,2,4,1};		//here we already created an array of size 6 instead of taking input from user
		int i ;
		printf("Your unsorted array is :  ");
		for (i=0;i<6;i++){
			printf("%d ",arr[i]);
		}
		int ub=5,lb=0;
		quicksort(arr,lb,ub);		//calling the sort function
		
		printf("\n\nYour sorted array is :  ");
		for (i=0;i<6;i++){
			printf("%d ",arr[i]);
		}	
}
									//swapping function defined
void swap(int arr[],int a,int b){
	int temp;
	temp=arr[a];
	arr[a]=arr[b];
	arr[b]=temp;
}
//Partition function defined
int partition(int arr[], int lb, int ub){
	int pivot=arr[lb];
	int start=lb;
	int	end=ub;
	
	while(start<end){
		while(arr[start]<=pivot){
			start++;
		}
		while(arr[end]>pivot){
			end--;
		}
		if(start<end){
		swap(arr,start,end);
		}
	}
	swap(arr,end,lb);
	return end;
}
// sorting function defined
void quicksort(int arr[],int lb,int ub){
	int d;
	if (lb<ub){
		d=partition(arr,lb,ub);
		quicksort(arr,lb,d-1);
		quicksort(arr,d+1,ub);
	}
}

File: basic data structure using python/quick sort.py
#here instead of takin input from user we created a list having ome data element:
a=[3,8,5,2,4,1]
print(f"The data before sorting is : {a}")
print("sorting initiated...........")
lb=0
ub=len(a)-1

#function for partition
def partition(arr,lb,ub):
    start=lb
    end=ub
    pivot=a[lb]
    while(start<end):
        while(a[start]<=pivot):
            start=start+1
        while(a[end]>pivot):
            end=end-1
        if(start<end):
            a[start],a[end]=a[end],a[start]

    a[lb],a[end]=a[end],a[lb]
    return end

#function is defined here
def quicksort(a,lb,ub):
    if(lb<ub):
        d=partition(a,lb,ub)
        quicksort(a,lb,d-1)
        quicksort(a,d+1,ub)

print("\n")
#function is called here.
quicksort(a,lb,ub)
print(f"The sorted data is : {a}")
